## Problem 1
    Write a program that will calculate the in-hand monthly salary after deductions on CTC:
    HRA = 10%
    DA = 5%
    PF = 3%
    Taxes deduction:
    Below 5 Lakhs → 0%
    5–10 Lakhs → 10%
    10–20 Lakhs → 20%
    Above 20 Lakhs → 3

In [4]:
ctc = float(input("Enter your CTC in lakhs : "))

annual_salary = ctc*100000

hra = 0.10 * annual_salary
da = 0.05 * annual_salary
pf = 0.03 * annual_salary 

if ctc <= 5:
    tax_percent = 0 
elif ctc <= 10:
    tax_percent = 10 
elif ctc <= 20:
    tax_percent = 20
else:
    tax_percent = 30

tax_amount = (tax_percent / 100) * annual_salary 

inhand_annual = annual_salary - (hra + da + pf + tax_amount)

monthly_salary = inhand_annual / 12

print("---- Salary Breakdown ----")
print(f"Annual salary : {annual_salary}")
print(f"HRA Deduction (10%) : {hra}")
print(f"DA Deduction (5%) : {da}")
print(f"PF Deduction (3%) : {pf}")
print(f"TAX ({tax_percent}% : {tax_amount})")
print(f"In Hand Monthly Salary : {monthly_salary}")

Enter your CTC in lakhs :  10


---- Salary Breakdown ----
Annual salary : 1000000.0
HRA Deduction (10%) : 100000.0
DA Deduction (5%) : 50000.0
PF Deduction (3%) : 30000.0
TAX (10% : 100000.0)
In Hand Monthly Salary : 60000.0


## Problem 2
    Write a program that takes user input of three angles and determines whether they can form a triangle.

In [2]:
a = float(input("Enter first angle : "))
b = float(input("Enter second angle : "))
c = float(input("Enter third angle : "))
d = a + b + c

if a <= 0 or b <= 0 or c <= 0 :
    print("Angles cant be 0 or negative.")
elif d == 180:
    print("These angles can form a triangle.")
else:
    print("These angles cannot form a triangle.")

Enter first angle :  50
Enter second angle :  40
Enter third angle :  90


These angles can form a triangle.


## Problem 3
    Write a program that takes user input of cost price and selling price and determines whether it is a loss or a profit.

In [10]:
cp = float(input("Enter cost price: "))
sp = float(input("Enter selling price: "))

p = sp - cp
l = cp - sp

if sp > cp:
    print(f"It is a profit of {p} rs")
elif sp < cp:
    print(f"It is a loss of {l} rs")
else:
    print("No profit, no loss.")

Enter cost price:  100
Enter selling price:  120


It is a profit of 20.0 rs


## Problem 4: Menu Driven Program:
    cm to ft
    km to miles
    USD to INR

In [12]:
while True:
    print("\n1. Convert cm to ft")
    print("2. Convert km to miles")
    print("3. Convert USD to INR")
    print("4. Exit")

    choice = input("Enter your choice (1/2/3/4): ")

    if choice == "1":
        cm = float(input("Enter value in cm: "))
        ft = cm / 30.48
        print("Feet:", ft)

    elif choice == "2":
        km = float(input("Enter value in km: "))
        miles = km * 0.621371
        print("Miles:", miles)

    elif choice == "3":
        usd = float(input("Enter amount in USD: "))
        inr = usd * 83.0     
        print("INR:", inr)

    elif choice == "4":
        print("Exit program")
        break

    else:
        print("Invalid choice")



1. Convert cm to ft
2. Convert km to miles
3. Convert USD to INR
4. Exit


Enter your choice (1/2/3/4):  1
Enter value in cm:  1000


Feet: 32.808398950131235

1. Convert cm to ft
2. Convert km to miles
3. Convert USD to INR
4. Exit


Enter your choice (1/2/3/4):  2
Enter value in km:  120


Miles: 74.56452

1. Convert cm to ft
2. Convert km to miles
3. Convert USD to INR
4. Exit


Enter your choice (1/2/3/4):  3
Enter amount in USD:  90


INR: 7470.0

1. Convert cm to ft
2. Convert km to miles
3. Convert USD to INR
4. Exit


Enter your choice (1/2/3/4):  4


Exiting program...


## Problem 5 : Fabonacci series 
    Display Fibonacci series up to 10 terms.
    First two numbers are 0 and 1.

In [13]:
a = 0
b = 1
print(a, b, end=" ")

for i in range(8):   # because 2 are already taken 0 and 1
    c = a + b
    print(c, end=" ")
    a = b
    b = c

0 1 1 2 3 5 8 13 21 34 

## Problem 6 Factorial
    Write a program to calculate the factorial of a number using a loop.
    Example:
    5! = 5 × 4 × 3 × 2 × 1 = 120

In [14]:
n = int(input("Enter a number: "))
fact = 1

for i in range(1, n + 1):
    fact *= i

print("Factorial =", fact)

Enter a number:  5


Factorial = 120


## Problem 7 Reverse an Integer
    Example:
    Input: 76542
    Output: 24567

In [15]:
num = int(input("Enter a number: "))
rev = 0

while num > 0:
    digit = num % 10
    rev = rev * 10 + digit
    num //= 10

print("Reversed number:", rev)

Enter a number:  76542


Reversed number: 24567


## Problem 8 
    Take user input N.
    Compute the sum from 1 to N with conditions:
      Skip numbers divisible by 5
      Stop summing if total exceeds 300
     Do NOT use a for loop

    Example:
    Input: 30
    Output: 276

In [18]:
n = int(input("Enter N: "))
total = 0
i = 1
print("Numbers added:", end=" ")

while i <= n:
    if i % 5 != 0:          
        if total + i > 300: 
            break
        print(i, end=" ")
        total += i
    i += 1

print("\nSum =", total)

Enter N:  30


Numbers added: 1 2 3 4 6 7 8 9 11 12 13 14 16 17 18 19 21 22 23 24 26 
Sum = 276


## Problem 9
    Write a program that keeps accepting numbers until the user enters 0.
    Display the sum and average of all values.

In [19]:
total = 0
count = 0

while True:
    num = float(input("Enter a number (0 to stop): "))
    if num == 0:
        break
    total += num
    count += 1

if count > 0:
    avg = total / count
    print("Sum =", total)
    print("Average =", avg)
else:
    print("No numbers were entered.")

Enter a number (0 to stop):  4
Enter a number (0 to stop):  3
Enter a number (0 to stop):  5
Enter a number (0 to stop):  1
Enter a number (0 to stop):  0


Sum = 13.0
Average = 3.25


## Problem 10 
    Find all numbers divisible by 7 but not multiples of 5, between 2000 and 3200 (inclusive).
    Print them as a comma-separated sequence.

In [21]:
result = []

for i in range(2000, 3201):
    if i % 7 == 0 and i % 5 != 0:
        result.append(str(i))

for i in range(0, len(result), 10):
    print(", ".join(result[i:i+10]))


2002, 2009, 2016, 2023, 2037, 2044, 2051, 2058, 2072, 2079
2086, 2093, 2107, 2114, 2121, 2128, 2142, 2149, 2156, 2163
2177, 2184, 2191, 2198, 2212, 2219, 2226, 2233, 2247, 2254
2261, 2268, 2282, 2289, 2296, 2303, 2317, 2324, 2331, 2338
2352, 2359, 2366, 2373, 2387, 2394, 2401, 2408, 2422, 2429
2436, 2443, 2457, 2464, 2471, 2478, 2492, 2499, 2506, 2513
2527, 2534, 2541, 2548, 2562, 2569, 2576, 2583, 2597, 2604
2611, 2618, 2632, 2639, 2646, 2653, 2667, 2674, 2681, 2688
2702, 2709, 2716, 2723, 2737, 2744, 2751, 2758, 2772, 2779
2786, 2793, 2807, 2814, 2821, 2828, 2842, 2849, 2856, 2863
2877, 2884, 2891, 2898, 2912, 2919, 2926, 2933, 2947, 2954
2961, 2968, 2982, 2989, 2996, 3003, 3017, 3024, 3031, 3038
3052, 3059, 3066, 3073, 3087, 3094, 3101, 3108, 3122, 3129
3136, 3143, 3157, 3164, 3171, 3178, 3192, 3199


## Problem 11 
    Print all numbers between 1000 and 3000 (inclusive) where every digit is even.
    Output must be space-separated.

In [23]:
result = []

for num in range(1000, 3001):
    s = str(num)
    all_even = True

    for digit in s:
        if int(digit) % 2 != 0:
            all_even = False
            break

    if all_even:
        result.append(str(num))

print(" ".join(result))


2000 2002 2004 2006 2008 2020 2022 2024 2026 2028 2040 2042 2044 2046 2048 2060 2062 2064 2066 2068 2080 2082 2084 2086 2088 2200 2202 2204 2206 2208 2220 2222 2224 2226 2228 2240 2242 2244 2246 2248 2260 2262 2264 2266 2268 2280 2282 2284 2286 2288 2400 2402 2404 2406 2408 2420 2422 2424 2426 2428 2440 2442 2444 2446 2448 2460 2462 2464 2466 2468 2480 2482 2484 2486 2488 2600 2602 2604 2606 2608 2620 2622 2624 2626 2628 2640 2642 2644 2646 2648 2660 2662 2664 2666 2668 2680 2682 2684 2686 2688 2800 2802 2804 2806 2808 2820 2822 2824 2826 2828 2840 2842 2844 2846 2848 2860 2862 2864 2866 2868 2880 2882 2884 2886 2888


## Problem 12: Robot Movement
    Robot moves from (0,0) using commands:
    UP X DOWN X LEFT X RIGHT X !
    ! means stop.
    Calculate the distance from original point.
    If the result is float, print nearest integer.
    Example:
    Input: UP 5 DOWN 3 LEFT 3 RIGHT 2 !
    Output: 2

In [1]:
x = 0
y = 0

directions = ["UP", "DOWN", "LEFT", "RIGHT"]
moves = int(input("How many moves? "))

for i in range(moves):
    direction = directions[i % 4]   # auto direction rotation
    print(f"Move {i+1}: {direction}")
    
    steps = int(input("Enter steps: "))

    if direction == "UP":
        y += steps
    elif direction == "DOWN":
        y -= steps
    elif direction == "LEFT":
        x -= steps
    elif direction == "RIGHT":
        x += steps

distance = (x**2 + y**2)**0.5
print("Distance from origin:", round(distance))

How many moves?  5


Move 1: UP


Enter steps:  4


Move 2: DOWN


Enter steps:  3


Move 3: LEFT


Enter steps:  5


Move 4: RIGHT


Enter steps:  6


Move 5: UP


Enter steps:  3


Distance from origin: 4


## Problem 13 : Prime Number Check
    Write a program to check whether a given number is prime.

In [3]:
n = int(input("Enter a number: "))

if n <= 1:
    print("Not a prime number")
else:
    is_prime = True
    i = 2

    while i * i <= n:
        if n % i == 0:
            is_prime = False
            break
        i += 1

    if is_prime:
        print("Prime number")
    else:
        print("Not a prime number")

Enter a number:  7


Prime number


## Problem 14 : Armstrong Numbers in a Range
    Print all Armstrong numbers in a user-given range.
    (Armstrong numbers: 0, 1, 153, 370, 371, 407…)

In [7]:
start = int(input("Enter start of range: "))
end = int(input("Enter end of range: "))

print("Armstrong numbers:")

for num in range(start, end + 1):
    digits = str(num)
    power = len(digits)
    total = sum(int(d)**power for d in digits)

    if (num == 1 or num >= 100) and total == num:
        print(num, end=" ")

Enter start of range:  1
Enter end of range:  500


Armstrong numbers:
1 153 370 371 407 

## Problem 15 : Clock Hand Angle
    Calculate the minimum angle between hour hand and minute hand.
    Print the floor of the final angle.

    Example:
    Input: H = 9, M = 0
    Output: 90

In [10]:
h = int(input("Enter hour: "))
m = int(input("Enter minutes: "))

h = h % 12

minute_angle = m * 6          
hour_angle = (h * 30) + (m * 0.5)   

angle = abs(hour_angle - minute_angle)

min_angle = min(angle, 360 - angle)

print(int(min_angle))   


Enter hour:  6
Enter minutes:  0


180


## Problem 16 : Rectangle Overlap
    Given two rectangles represented by coordinates of:

    Left-top corner
    Right-bottom corner
    Check whether the two rectangles overlap.
    Rectangles sharing a side are considered overlapping.
    Assume rectangles are parallel to the axes.

In [11]:
x1, y1 = map(float, input("Rect1 - enter x1 y1: ").split())
x2, y2 = map(float, input("Rect1 - enter x2 y2: ").split())

x3, y3 = map(float, input("Rect2 - enter x3 y3: ").split())
x4, y4 = map(float, input("Rect2 - enter x4 y4: ").split())

l1, r1 = min(x1, x2), max(x1, x2)
b1, t1 = min(y1, y2), max(y1, y2)

l2, r2 = min(x3, x4), max(x3, x4)
b2, t2 = min(y3, y4), max(y3, y4)

h_overlap = max(l1, l2) <= min(r1, r2)
v_overlap = max(b1, b2) <= min(t1, t2)

if h_overlap and v_overlap:
    print("Rectangles overlap")
else:
    print("Rectangles do not overlap")


Rect1 - enter x1 y1:  0 10
Rect1 - enter x2 y2:  10 0
Rect2 - enter x3 y3:  10 5
Rect2 - enter x4 y4:  20 -5


Rectangles overlap
